code/modules/power/battery_ion.dm

```diff
@@ -0,0 +1,70 @@
+// Generic battery machine
+// stores power
+
+/obj/machinery/power/battery/portable
+	name = "portable power storage unit"
+	desc = "A IOn-model portable storage unit, used to transport charge around the station."
+	icon_state = "port_smes"
+	density = 1
+	anchored = 0
+	use_power = 0
+
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE
+
+	var/obj/machinery/power/battery_port/connected_to
+
+/obj/machinery/power/battery/portable/initialize()
+	..()
+	if(anchored)
+		var/obj/machinery/power/battery_port/port = locate() in src.loc
+		if(port)
+			port.connect_battery(src)
+
+/obj/machinery/power/battery/portable/get_powernet()
+	if(connected_to)
+		return connected_to.get_powernet()
+
+/obj/machinery/power/battery/portable/add_avail(var/amount)
+	if(connected_to)
+		connected_to.add_avail(amount)
+
+/obj/machinery/power/battery/portable/add_load(var/amount)
+	if(connected_to)
+		connected_to.add_load(amount)
+
+/obj/machinery/power/battery/portable/surplus()
+	if(connected_to)
+		return connected_to.surplus()
+	return 0
+
+/obj/machinery/power/battery/portable/wrenchAnchor(mob/user)
+	if(..() == 1)
+		if(anchored)
+			var/obj/machinery/power/battery_port/port = locate() in src.loc
+			if(port)
+				port.connect_battery(src)
+		else
+			if(connected_to)
+				connected_to.disconnect_battery()
+		return 1
+	return -1
+
+/obj/machinery/power/battery/portable/update_icon()
```

Comment:
This method of doing images is bad. Take the method APCs and now canister's do, it simply initializes a global list holding the image objects that can be applied to your object, and then applies the same image to any instance of this object in the overlays preventing the need for any new images or deleted images.